### PR TITLE
OGD-647: add custom dimension tracking

### PR DESCRIPTION
**Pull request by psavary**

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -39,6 +43,18 @@
             _paq.push(['trackEvent', action, origin, target]);
         }
 
+        function trackOrganization(organization, action, dataset, format) {
+            //todo track view when dataset page is rendered
+            customDimensionDataset = {};
+            customDimensionDataset['dimension'+customDimensionActionDatasetId] = organization;
```
Comment (tschortsch) [inline assets/js/app.js]: Where does the `customDimensionActionDatasetId` variable come from?

File: assets/js/app.js [hunk 2]
```diff
@@ -39,6 +43,18 @@
...
+            customDimensionFormat = {};
+            customDimensionFormat['dimension'+customDimensionActionFormatId] = organization;
```
Comment (tschortsch) [inline assets/js/app.js]: Shouldn't this be `customDimensionFormat['dimension'+customDimensionActionFormatId] = format;` ?

File: assets/js/app.js [hunk 3]
```diff
@@ -39,6 +43,18 @@
...
+            _paq.push(['trackEvent', 'click', action, dataset, '', customDimensionDataset]);
+            _paq.push(['trackEvent', 'click', action, format, '', customDimensionFormat]);
+        }
+
+
```
Comment (tschortsch) [inline assets/js/app.js]: Please remove unneeded line breaks

**Review by tschortsch — CHANGES_REQUESTED** [review]

File: assets/js/app.js (hunk 1 above)
Comment (psavary) [reply assets/js/app.js]: Initialized in https://github.com/opendata-swiss/ckanext-switzerland/pull/167/files

File: assets/js/app.js (hunk 3 above)
Comment (psavary) [reply assets/js/app.js]: done

File: assets/js/app.js (hunk 2 above)
Comment (psavary) [reply assets/js/app.js]: no, we push the format on the trackEvent as name:
https://developer.matomo.org/guides/tracking-javascript-guide#custom-dimensions

File: assets/js/app.js (hunk 1 above)
Comment (tschortsch) [reply assets/js/app.js]: I see, but then please make sure that they are initialized before using them.

Comment (psavary) [reply assets/js/app.js]: OK, We load the app.js after the variables in https://github.com/opendata-swiss/ckanext-switzerland/pull/167/files/ed1948c7c655da90f72243067b929a9d7cfabc38#diff-86553814cbab74f96827be9c81acea6d
is that sufficient?

Comment (tschortsch) [reply assets/js/app.js]: Not really since the app.js is also used in WordPress where these variables are never initialized.

**Review by tschortsch — APPROVED** [review]

**Pull request merged**